mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,443 @@
+                        self._log_ckpt_as_artifact(self.ckpt_hook.out_dir,
+                                                   runner.epoch, aliases)
+
+        if self.num_eval_images > 0 and self.log_evaluation:
+            if self.eval_hook.by_epoch and self.eval_hook._should_evaluate(
+                    runner):
+                results = self.eval_hook.results
+                # Initialize evaluation table
+                self._init_pred_table()
+                # Log predictions
+                self._log_predictions(results, runner.epoch + 1)
+                # Log the table
+                self._log_eval_table()
+
+    @master_only
+    def after_run(self, runner):
+        self.wandb.finish()
+
+    def _log_ckpt_as_artifact(self,
+                              path_to_model,
+                              epoch,
+                              aliases,
+                              metadata=None):
+        """Log model checkpoint as  W&B Artifact.
+
+        Args:
+            path_to_model (str): Path where model checkpoints are saved.
+            epoch (int): The current epoch.
+            aliases (list): List of the aliases associated with this artifact.
+            metadata (dict, optional): Metadata associated with this artifact.
+        """
+        model_artifact = self.wandb.Artifact(
+            f'run_{self.wandb.run.id}_model', type='model', metadata=metadata)
+        model_artifact.add_file(f'{path_to_model}/epoch_{epoch+1}.pth')
+        self.wandb.log_artifact(model_artifact, aliases=aliases)
+
+    def _get_ckpt_metadata(self, runner):
+        """Get model checkpoint metadata."""
+        if self.ckpt_hook.interval == self.eval_hook.interval:
+            results = self.eval_hook.results
+        else:
+            runner.logger.info(
+                f'Evaluating for model checkpoint at epoch '
+                f'{runner.epoch+1} which will be saved as W&B Artifact.')
+            if isinstance(self.eval_hook, EvalHook):
+                from mmdet.apis import single_gpu_test
+                results = single_gpu_test(
+                    runner.model, self.val_dataloader, show=False)
+            elif isinstance(self.eval_hook, DistEvalHook):
+                from mmdet.apis import multi_gpu_test
+                results = multi_gpu_test(
+                    runner.model, self.val_dataloader, gpu_collect=True)
+
+        eval_results = self.val_dataset.evaluate(results, logger='silent')
+        metadata = dict(epoch=runner.epoch + 1, **eval_results)
+        return metadata
+
+    def _is_best_ckpt(self, metadata):
+        """Check if the current checkpoint has the best metric score.
+
+        Args:
+            metadata (dict): Metadata associated with the checkpoint.
+
+        Returns:
+            bool: Returns True, if the checkpoint has the best metric score.
+        """
+        keys = list(metadata.keys())
+        map_metrics = [key for key in keys if 'mAP' in key]
```

Comment:
This function cannot handle panoptic segmentation, and there is no generality. My suggestion is that it does not need to support this function.

Reply:
Is there a way to programmatically find the evaluation metrics? Or is there a set of evaluation metrics upon which we can determine which checkpoint is the best one? 

I personally find it useful that in the UI, I have the information about the best checkpoint.

Reply:
There is no particularly good way for the time being, if we must support it, we can only hard code.

Reply:
I have removed this for now.